app/javascript/articles/Feed.jsx

```diff
@@ -0,0 +1,138 @@
+import { h, Component } from 'preact';
+import PropTypes from 'prop-types';
+import { FEED_ICONS } from '../packs/feedIcons.js.erb';
+
+/* global userData sendHapticMessage showModal buttonFormData renderNewSidebarCount */
+
+export class Feed extends Component {
+  componentDidMount() {
+    const { timeFrame } = this.props;
+    const { reading_list_ids = [] } = userData(); // eslint-disable-line camelcase
+
+    this.setState({ bookmarkedFeedItems: new Set(reading_list_ids) });
```

Comment:
I went with a `Set` because the current code was an array of IDs which meant that the operation to find out if an article is in the reading list was n<sup>2<sup>. Now it is log n.